product/roundhouse.tasks/Roundhouse.cs

```diff
@@ -142,12 +153,14 @@ public string RecoveryMode
             set
             {
                 RecoveryMode result;
-                if (!Enum.TryParse(value, true, out result))
+                if (Enum.TryParse(value, true, out result))
                 {
-                    throw new ArgumentOutOfRangeException("value", value, "The value of 'RecoveryMode' must be one of these values: 'NoChange', 'Simple' or 'Full'.");
+                    this.recoveryMode = result;
+                }
+                else
+                {
+                    this.loggingHelper.LogError("The value of 'RecoveryMode' must be one of these values: 'NoChange', 'Simple' or 'Full'. Actual value was '{0}'.", value);

```

Comment:
The simpler way is to set a string to an error if you are unable to set and then write the error out in the execute task. Boom, no need for adding msbuild tasks and taking over the loggers.